lib/shopify-cli/theme/syncer.rb

```diff
@@ -277,7 +277,7 @@ def delete(file)
 
       def update_checksums(api_response)
         api_response.values.flatten.each do |asset|
-          if asset["key"] && asset["checksum"]
+          if asset["key"]
             @checksums[asset["key"]] = asset["checksum"]
```

Comment:
Does setting `@checksums[asset["key"]] = nil` (in the case where `asset["checksum"]` is `nil`, which was filtered by the previous version of the `if` expression) cause ripple effects elsewhere, or is it safe?  The main use seems to be in `file_has_changed?`, so it seems OK to set the checksum to `nil`, but I'm not as familiar with the intention of this code.

Reply:
This is only used internally by this class. It should have no effects outside of fixing the issue w/ `theme pull`.

`@checksums[key]` will return `nil` either if the key is not present or if explicitly set to `nil`, so this method behavior will not change.